src/utils/autoprefixer.js

```diff
@@ -20,7 +20,14 @@ export default function(muiTheme) {
   if (userAgent === false) { // Disabled autoprefixer
     return null;
   } else if (userAgent === 'all' || userAgent === undefined) { // Prefix for all user agent
-    return (style) => InlineStylePrefixer.prefixAll(style);
+    return (style) => {
+      const isFlex = ['flex', 'inline-flex'].includes(style.display);
+      const o = InlineStylePrefixer.prefixAll(style);
+      if (isFlex) {
+        o.display = o.display.join('; display: ');
```

Comment:
That won't work with react-dom (https://github.com/facebook/react/issues/6467).
Still, that would be great to have this looking when doing SSR 👍 .
What about adding `if (typeof window === 'undefined') {`?